day-3/part-2/part-2.py

```diff
@@ -0,0 +1,88 @@
+import re
+
+GEAR = '*'
+DIGITS = [str(num) for num in range(10)]
+
+class Gear:
+    def __init__(self):
+        self.numbers = []
+    
+    def get_product(self):
+        if len(self.numbers) == 2:
+            return self.numbers[0] * self.numbers[1]
+        return 0
+
+class Number:
+    def __init__(self, column, line_start, line_end, value):
+        self.column = column
+        self.line_start = line_start
+        self.line_end = line_end
+        self.value = value
+
+    def is_touched_by_gear(self, column, line):
+        gear_adjacents = self.get_adjacent_cells(line, column)
+        for line_index in range(self.line_start, self.line_end):
+            if (self.column, line_index) in gear_adjacents:
+                return True
+        return False
+
+    def get_adjacent_cells(self, line, column):
```

Comment:
I'm not a fan of this because it seems like you're calculating cells that are adjacent to a gear here, but this method is inside the Number class. So it's misleading when you call `number.get_adjacent_cells()`.

Reply:
True! Just fixed it.